File: src/kyu_7/all_unique/c/all_unique.c
#include <stdbool.h>
#include <string.h>

bool has_unique_chars(const char *str) 
{
    char c, tab[128] = { 0 };
    while (c = *str++)
    {
        if (tab[c]++)
        {
           return false;
        }
    }
    return true;
}

File: src/kyu_7/all_unique/csharp/all_unique.cs
using System.Collections.Generic;

namespace Code
{
  public class Unique
  {
    public static bool HasUniqueChars(string str)
    {
        HashSet<char> prepared = new HashSet<char>();
        for (int i = 0; i < str.Length; i++)
        {
            prepared.Add(str[i]);
        }
        return prepared.Count == str.Length;
    }
  }
}